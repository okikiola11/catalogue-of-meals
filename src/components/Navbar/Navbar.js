import React from 'react';
import { Link } from 'react-router-dom';

import './Navbar.module.css';

const Navbar = () => (
  <header className="text-center">
    <nav className="bg-color">
      <Link to="/"> Meals </Link>
    </nav>
  </header>
);

export default Navbar;
