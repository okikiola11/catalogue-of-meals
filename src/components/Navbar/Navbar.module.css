@import url('https://fonts.googleapis.com/css2?family=Cookie&display=swap');

header {
  height: 80px;
  box-shadow: 0 10px 6px -6px #777;
  font-family: 'Cookie', cursive;
}

h1 {
  line-height: 80px;
}