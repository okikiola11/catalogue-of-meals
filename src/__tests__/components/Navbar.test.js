import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';

import Navbar from '../../components/Navbar/Navbar';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Navbar />, div);
});

it('matches Navbar snapshot', () => {
  const tree = renderer
    .create(
      <Router>
        <Navbar />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains Meals logo', () => {
  render(
    <Router>
      <Navbar />
    </Router>,
  );
  expect(screen.getByText('Meals')).toBeInTheDocument();
});
