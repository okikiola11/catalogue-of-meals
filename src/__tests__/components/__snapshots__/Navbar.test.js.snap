// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Navbar snapshot 1`] = `
<header
  className="text-center bg-color"
>
  <nav
    className=""
  >
    <a
      href="/"
      onClick={[Function]}
    >
       Meals 
    </a>
  </nav>
</header>
`;
